js/index.js

```diff
@@ -1,42 +1,101 @@
 // ITERATION 1
 
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
+  const priceElement = product.querySelector('.price span');
+  const quantityElement = product.querySelector('.quantity input');
+  const subtotalElement = product.querySelector('.subtotal span');
 
-  //... your code goes here
+  const priceValue = priceElement.innerText;
+  const quantityValue = quantityElement.value;
+  const subtotalPrice = (priceValue * quantityValue).toFixed(2);
+
+  subtotalElement.innerText = subtotalPrice;
+  return subtotalPrice;
 }
 
 function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
+  const products = document.querySelectorAll('.product');
+  [...products].map(p => updateSubtotal(p));
 
-  // ITERATION 2
-  //... your code goes here
+  const subtotals = document.querySelectorAll('.subtotal span');
+  const totalPrice = [...subtotals].map(s => parseFloat(s.innerText))
+    .reduce((a, v) => a + v);
 
-  // ITERATION 3
-  //... your code goes here
+  document.querySelector('#total-value span').innerText = totalPrice.toFixed(2);
 }
 
 // ITERATION 4
 
 function removeProduct(event) {
-  const target = event.currentTarget;
-  console.log('The target in remove is:', target);
-  //... your code goes here
+  const target = event.target;
+  const product = target.parentElement.parentElement;
+  const productsTable = product.parentElement;
+  productsTable.removeChild(product);
 }
```

Comment:
You could call here the function `calculateAll()` to recalculate the total after removing a product.